ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/legaladvisor/CCDHearingCourtType.java

```diff
@@ -1,11 +1,21 @@
 package uk.gov.hmcts.cmc.ccd.domain.legaladvisor;
 
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+
+@AllArgsConstructor
+@NoArgsConstructor
 public enum CCDHearingCourtType {
-    EDMONTON,
-    MANCHESTER,
-    BIRMINGHAM,
-    CLERKENWELL,
-    SHOREDITCH,
+    EDMONTON("N182TN"),
+    MANCHESTER("M609DJ"),
+    BIRMINGHAM("B11AA"),
+    CLERKENWELL("EC1V3RE"),
     DEFENDANT_COURT,
-    CLAIMANT_COURT
+    CLAIMANT_COURT;
+
+    private String postcode;
+
+    public String getPostcode() {
```

Comment:
How about returning `Optional<String> getPostcode()` here ?

Reply:
I had done it like that originally, but it really adds no benefit as we already "know" which ones have a postcode and which ones don't.
This enum is weird in the first place, as it mixes up "hardcoded" values with dynamic ones (CLAIMANT vs DEFENDANT).

Reply:
With the current values in enum and it's usage it looks fine. 

There will be issue if we add another enum (with null postcode) here and don't add that in switch here https://github.com/hmcts/cmc-claim-store/pull/925/files#diff-c6dc194a2b0e1cdf996662e0fd7181baR101 

I will leave that with you to decide. 👍